core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -338,7 +338,7 @@ private[spark] class TaskSetManager(
       if (TaskLocality.isAllowed(locality, TaskLocality.RACK_LOCAL)) {
         for (rack <- sched.getRackForHost(host)) {
           for (index <- speculatableTasks if canRunOnHost(index)) {
```

Comment:
not a big deal but I think that can happen outside of this patch since it's just cleanup